src/rejectdups-rust/src/lib.rs

```diff
@@ -0,0 +1,500 @@
+use byteorder::{ByteOrder, LittleEndian};
+use solana_sdk::{
+    account_info::AccountInfo, entrypoint, entrypoint::ProgramResult, info,
+    program_error::ProgramError, account_info::next_account_info, pubkey::Pubkey,
+    hash::Hasher, 
+    sysvar::{self, Sysvar}, rent::Rent,
+};
+use std::mem;
+
+
+pub fn create_address_with_seed(
+    base: &Pubkey,
+    seed: &str,
+    program_id: &Pubkey,
+) -> Pubkey {
+    let mut hasher = Hasher::default();
+    hasher.hashv(&[base.as_ref(), seed.as_ref(), program_id.as_ref()]);
+    Pubkey::new( hasher.result().as_ref() )
+}
+
+// Declare and export the program's entrypoint
+entrypoint!(process_instruction);
+
+// Program entrypoint's implementation
+fn process_instruction<'a>(
+    program_id: &Pubkey,             // Public key of program account
+    accounts: &'a [AccountInfo<'a>], // data accounts
+    instruction_data: &[u8],         // 1 = vote for A, 2 = vote for B
+) -> ProgramResult {
+
+    info!("Rust program entrypoint");
+
+    // Iterating accounts is safer then indexing
+    let accounts_iter = &mut accounts.iter();
+
+    // Get the account that holds the vote count
+    let account = next_account_info(accounts_iter)?;
+
+    // The account must be owned by the program in order to modify its data
+    if account.owner != program_id {
+        info!("Vote account does not have the correct program id");
+        return Err(ProgramError::IncorrectProgramId);
```

Comment:
The program id might not be the issue here, this error could be described as the wrong owner.  Could define your own error values here to more accurately describe the error encountered.

Reply:
ok, i make custom error